### PR TITLE
Metrics-based SLOs (GA 8/18)

### PR DESCRIPTION
**Pull request by kimsauce**
## Purpose of this pull request

Ability to create SLO from Metrics query (GA 8/18)

## Select the type of change:

- [ ] Minor Changes - Typos, formatting, slight revisions
- [ ] Update Content - Revisions and updating sections
- [x] New Content - New features, sections, pages, tutorials
- [ ] Site and Tools - Updates, maintenance, and new packages for the site, Gatsby, React, etc

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204737230246488
  - https://app.asana.com/0/0/1204737230246486

### CODE REVIEW

File: blog-service/2023-08-09-metrics-slos.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  - observability
+  - metrics
+  - operators
+hide_table_of_contents: true
+authors:
+  - url: https://help.sumologic.com/release-notes-service/rss.xml
+    image_url: /img/release-notes/rss-orange.png
+---
+
+You can now create SLOs from the **Metrics** page and use [metrics operators](/docs/metrics/metrics-operators) in these SLO queries. The metrics query specified in your SLO should be a pure selector query. You can specify one or more operators in the query for SLO.
```
Comment (tarunk2) [inline blog-service/2023-08-09-metrics-slos.md:15]: `You can now create SLOs from the **Metrics** page` - this is already supported. This change is for supporting operators in metrics-based SLOs.

Comment (tarunk2) [reply blog-service/2023-08-09-metrics-slos.md:15]: `The metrics query specified in your SLO should be a pure selector query.` - This is in conflict with this change and not correct.

File: blog-service/2023-08-09-metrics-slos.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+[Learn more]
+
+NOTE: <!-- Does this mean not available to all customers yet? Or not all metrics operators available yet? -->
```
Comment (tarunk2) [inline blog-service/2023-08-09-metrics-slos.md:19]: This feature will be available to all customers. All metrics operators will be supported as well.

**Review by tarunk2 — CHANGES_REQUESTED** [review]

File: docs/observability/reliability-management-slo/create-slo.md [hunk 3]
```diff
@@ -307,8 +307,16 @@ You can make further modifications to a saved filter view later using kebab menu
...
-:::note Coming soon
-You'll be able to create SLOs from Metrics Explorer.
+To create an SLO from the **Metrics** page:
+
+1. Click xyz...
+
+You can use [metrics operators](/docs/metrics/metrics-operators) for metrics-based SLOs. The metrics query specified in your SLO should be a pure selector query. You can specify one or more operators in the query for SLO.
+
+As an example, a pure selector query with no operators is `_sourceHost=stag-monitor-manager metric=CPU_Total`. That same query using the `sum` operator would be `_sourceHost=stag-monitor-manager metric=CPU_Total | sum`.
+
```
Comment (jinkane) [inline docs/observability/reliability-management-slo/create-slo.md:317]: ```That same query using the `sum` operator would be `_sourceHost=stag-monitor-manager metric=CPU_Total | sum`.```

To use operator we have a constraint that the user should provide a quantization after the selector. Example: `_sourceHost=stag-monitor-manager metric=CPU_Total | quantize to 1m using avg | sum`

File: blog-service/2023-08-09-metrics-slos.md (hunk 1 above)
Comment (kimsauce) [reply blog-service/2023-08-09-metrics-slos.md:15]: @tarunk2 @jinkane ,  thank you both for the clarification! Can you please provide what should be the correct edits in suggest mode? Then I'll accept the comments and we can publish.

https://github.com/SumoLogic/sumologic-documentation/assets/56411016/d0213a22-aef8-4a93-aeae-c885979f5b8e

File: docs/observability/reliability-management-slo/create-slo.md [hunk 4]
```diff
@@ -296,7 +296,7 @@ You can see the list of all saved filter views by clicking on the funnel icon.
 
 <img src={useBaseUrl('img/observability/ListOfSavedFilters.png')} alt="ListOfSavedFilters.png" />
 
-You can make further modifications to a saved filter view later using kebab menu options next to the funnel icon. 
+You can make further modifications to a saved filter view later using kebab menu options next to the funnel icon.
```
Comment (kimsauce) [inline docs/observability/reliability-management-slo/create-slo.md:299]:
```suggestion
You can make further modifications to a saved filter view later using kebab menu options next to the funnel icon. 
```

File: docs/observability/reliability-management-slo/create-slo.md [hunk 5]
```diff
@@ -307,9 +307,13 @@ You can make further modifications to a saved filter view later using kebab menu
 
 ## Create an SLO from Metrics page
 
-:::note Coming soon
-You'll be able to create SLOs from Metrics Explorer.
-:::
+To create an SLO from the **Metrics** page:
+
+1. Click xyz...
```
Comment (kimsauce) [inline docs/observability/reliability-management-slo/create-slo.md:312]: You can create SLOs with operators, provided you use an appropriate quantization.

```suggestion
1. Go to **Manage Data** > **Metrics**.
1. Under **Metrics Explorer**, select your desired **Metric** and **Filters**. Optionally, you can **Add Operator**.
1. Click the three-dot kebab icon, then select **Create SLO**.
1. Follow the instructions under [Create an SLO (General)](#create-an-slo-general).
```

File: docs/observability/reliability-management-slo/create-slo.md (hunk 4 above)
Comment (kimsauce) [inline docs/observability/reliability-management-slo/create-slo.md:299]:
```suggestion
You can make further modifications to a saved filter view later using kebab menu options next to the funnel icon. 
```

File: docs/observability/reliability-management-slo/create-slo.md (hunk 3 above)
Comment (kimsauce) [reply docs/observability/reliability-management-slo/create-slo.md:320]: https://github.com/SumoLogic/sumologic-documentation/pull/2725#discussion_r1289237320

File: blog-service/2023-08-18-metrics-slos.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  - observability
+  - metrics
+  - operators
+hide_table_of_contents: true
+authors:
+  - url: https://help.sumologic.com/release-notes-service/rss.xml
+    image_url: /img/release-notes/rss-orange.png
+---
+
+You can now create SLOs from the **Metrics** page and use [metrics operators](/docs/metrics/metrics-operators) in these SLO queries. The metrics query specified in your SLO should be a pure selector query. You can specify one or more operators in the query for SLO. [Learn more](/docs/observability/reliability-management-slo/create-slo/#create-an-slo-from-metrics-page).
```
Comment (jinkane) [inline blog-service/2023-08-18-metrics-slos.md:15]:
```suggestion
You can now create SLOs from the **Metrics** page and use [metrics operators](/docs/metrics/metrics-operators) in these SLO queries. The metrics query specified in your SLO should have a quantization specified after the selector. You can specify one or more operators in the query for SLO. [Learn more](/docs/observability/reliability-management-slo/create-slo/#create-an-slo-from-metrics-page).
```

File: docs/observability/reliability-management-slo/create-slo.md [hunk 7]
```diff
@@ -307,9 +307,16 @@ You can make further modifications to a saved filter view later using kebab menu
...
-You'll be able to create SLOs from Metrics Explorer.
-:::
+To create an SLO from the **Metrics** page:
+
+1. Go to **Manage Data** > **Metrics**.
+1. Under **Metrics Explorer**, select your desired **Metric** and **Filters**. Optionally, you can **Add Operator**.<br/><img src={useBaseUrl('img/observability/metrics-slo.png')} alt="metrics-slo.png" />
+1. Click the three-dot kebab icon, then select **Create SLO**.
+1. Follow the instructions under [Create an SLO (General)](#create-an-slo-general).
+
+You can use [metrics operators](/docs/metrics/metrics-operators) for metrics-based SLOs. The metrics query specified in your SLO should be a pure selector query. You can specify one or more operators in the query for SLO.
```
Comment (jinkane) [inline docs/observability/reliability-management-slo/create-slo.md:317]:
```suggestion
You can use [metrics operators](/docs/metrics/metrics-operators) for metrics-based SLOs. The metrics query specified in your SLO should be have a quantization after the selector. You can specify one or more operators in the query for SLO.
```

File: docs/observability/reliability-management-slo/create-slo.md [hunk 8]
```diff
@@ -307,9 +307,16 @@ You can make further modifications to a saved filter view later using kebab menu
...
+
+As an example, a pure selector query with no operators is `_sourceHost=stag-monitor-manager metric=CPU_Total`. That same query using the `sum` operator would be `_sourceHost=stag-monitor-manager metric=CPU_Total | sum`.
```
Comment (jinkane) [inline docs/observability/reliability-management-slo/create-slo.md:319]:
```suggestion
As an example, a pure selector query with no operators is `_sourceHost=stag-monitor-manager metric=CPU_Total`. That same query using the `sum` operator with appropriate quantization would be `_sourceHost=stag-monitor-manager metric=CPU_Total | quantize to 1m using avg | sum`.
```

**Review by jpipkin1 — APPROVED** [review]
Approved contingent on other comments being resolved.

File: blog-service/2023-08-18-metrics-slos.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+  - observability
+  - metrics
+  - operators
+hide_table_of_contents: true
+authors:
+  - url: https://help.sumologic.com/release-notes-service/rss.xml
+    image_url: /img/release-notes/rss-orange.png
+---
+
+You can now create SLOs from the **Metrics** page and use [metrics operators](/docs/metrics/metrics-operators) in these SLO queries. The metrics query specified in your SLO should have a quantization specified after the selector. You can specify one or more operators in the query for SLO. [Learn more](/docs/observability/reliability-management-slo/create-slo/#create-an-slo-from-metrics-page).
```
Comment (tarunk2) [inline blog-service/2023-08-18-metrics-slos.md:15]:
```suggestion
You can now create SLOs from the **Metrics** page with [metrics operators](/docs/metrics/metrics-operators) in SLO queries. The metrics query specified in your SLO should have a quantization specified after the selector. You can specify one or more operators in the query for SLO. [Learn more](/docs/observability/reliability-management-slo/create-slo/#create-an-slo-from-metrics-page).
```

File: docs/observability/reliability-management-slo/create-slo.md [hunk 10]
```diff
@@ -307,9 +307,16 @@ You can make further modifications to a saved filter view later using kebab menu
 
 ## Create an SLO from Metrics page
 
-:::note Coming soon
-You'll be able to create SLOs from Metrics Explorer.
-:::
+To create an SLO from the **Metrics** page:
+
+1. Go to **Manage Data** > **Metrics**.
+1. Under **Metrics Explorer**, select your desired **Metric** and **Filters**. Optionally, you can **Add Operator**.<br/><img src={useBaseUrl('img/observability/metrics-slo.png')} alt="metrics-slo.png" />
```
Comment (tarunk2) [inline docs/observability/reliability-management-slo/create-slo.md:313]: In this image, "Filters" params are not fully visible. Can we use advanced mode? That will make it easier to understand.

**Review by tarunk2 — COMMENTED** [review]
cc @apoorv-garg as well for the review.

File: docs/observability/reliability-management-slo/create-slo.md [hunk 11]
```diff
@@ -307,9 +307,16 @@ You can make further modifications to a saved filter view later using kebab menu
 
 ## Create an SLO from Metrics page
 
-:::note Coming soon
-You'll be able to create SLOs from Metrics Explorer.
-:::
+To create an SLO from the **Metrics** page:
+
+1. Go to **Manage Data** > **Metrics**.
```
Comment (apoorv-garg) [inline docs/observability/reliability-management-slo/create-slo.md:312]: this step is incorrect and can be removed

Comment (apoorv-garg) [inline docs/observability/reliability-management-slo/create-slo.md:312]:
```suggestion
1. Open a new or go to an existing **Metrics** tab.
```

File: docs/observability/reliability-management-slo/create-slo.md [hunk 12]
```diff
@@ -307,9 +307,16 @@ You can make further modifications to a saved filter view later using kebab menu
...
+1. Under **Metrics Explorer**, select your desired **Metric** and **Filters**. Optionally, you can **Add Operator**.<br/><img src={useBaseUrl('img/observability/metrics-slo.png')} alt="metrics-slo.png" />
+1. Click the three-dot kebab icon, then select **Create SLO**.
```
Comment (apoorv-garg) [inline docs/observability/reliability-management-slo/create-slo.md:314]:
```suggestion
1. Click the three-dot kebab icon, then select **Create an SLO**.
```

File: docs/observability/reliability-management-slo/create-slo.md [hunk 13]
```diff
@@ -307,9 +307,16 @@ You can make further modifications to a saved filter view later using kebab menu
...
+1. Follow the instructions under [Create an SLO (General)](#create-an-slo-general).
+
+You can use [metrics operators](/docs/metrics/metrics-operators) for metrics-based SLOs. The metrics query specified in your SLO should have a quantization after the selector. You can specify one or more operators in the query for SLO.
+
+As an example, a pure selector query with no operators is `_sourceHost=stag-monitor-manager metric=CPU_Total`. That same query using the `sum` operator with appropriate quantization would be `_sourceHost=stag-monitor-manager metric=CPU_Total | quantize to 1m using avg | sum`.
```
Comment (apoorv-garg) [inline docs/observability/reliability-management-slo/create-slo.md:319]:
```suggestion
As an example, a pure selector query with no operators could be `_sourceCategory=my-web-server metric=is_healthy` which returns a time series per instance your web server indicating if it is healthy or not (`1` or `0`). To count the number of instances that were healthy in a given minute, the `sum` operator can be used with an appropriate quantization method and interval as follows `_sourceCategory=my-web-server metric=is_healthy | quantize to 1m using max | sum`.

File: blog-service/2023-08-18-metrics-slos.md [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+  - observability
+  - metrics
+  - operators
+hide_table_of_contents: true
+authors:
+  - url: https://help.sumologic.com/release-notes-service/rss.xml
+    image_url: /img/release-notes/rss-orange.png
+---
+
+You can now create SLOs from the **Metrics** page with [metrics operators](/docs/metrics/metrics-operators) in SLO queries. The metrics query specified in your SLO should have a quantization specified after the selector. You can specify one or more operators in the query for SLO. [Learn more](/docs/observability/reliability-management-slo/create-slo/#create-an-slo-from-metrics-page).
```
Comment (apoorv-garg) [inline blog-service/2023-08-18-metrics-slos.md:15]:
```suggestion
You can now use [metrics operators](/docs/metrics/metrics-operators) in SLO queries. The metrics query specified in your SLO should have a quantization specified after the selector. You can specify one or more operators in the query for SLO. [Learn more](/docs/observability/reliability-management-slo/create-slo/#create-an-slo-from-metrics-page).
```

File: docs/observability/reliability-management-slo/create-slo.md (hunk 11 above)
Comment (kimsauce) [reply docs/observability/reliability-management-slo/create-slo.md:312]: @apoorv-garg is this what you meant? 

_Click **+ New** > **Metrics** or go to an existing **Metrics** tab._

Comment (apoorv-garg) [reply docs/observability/reliability-management-slo/create-slo.md:312]: Yes

File: docs/observability/reliability-management-slo/create-slo.md (hunk 10 above)
Comment (kimsauce) [reply docs/observability/reliability-management-slo/create-slo.md:313]: Does this new I've uploaded suffice?

**Pull request merged**